src/starks/verifier.rs

```diff
@@ -503,31 +503,36 @@ fn reconstruct_deep_composition_poly_evaluation<F: IsFFTField, A: AIR<Field = F>
     i: usize,
 ) -> FieldElement<F> {
     let primitive_root = &F::get_primitive_root_of_unity(domain.root_order as u64).unwrap();
+    // It's better to take this out of the function and perform batch inversion
     let upsilon_0 = &domain.lde_roots_of_unity_coset[iota_n];
 
     let mut trace_terms = FieldElement::zero();
-
+    let z_squared = &(&challenges.z * &challenges.z);
+    let den = (upsilon_0 - z_squared).inv();
```

Comment:
Use denominator instead of den, full name is better

Reply:
Already solved in [d257629](https://github.com/lambdaclass/lambdaworks_cairo_prover/pull/142/commits/d257629c4960bf20a56a1b7d78cd7dce63f3046e)